hsclient/hydroshare.py

```diff
@@ -1417,6 +1417,9 @@ def retrieve_zip(self, path, save_path="", params=None):
                 "Failed GET {}, status_code {}, message {}".format(url, response.status_code, response.content)
             )
         filename = path.split("/")[-1]
+        # append .zip to the filename if it doesn't end with .zip
+        if not filename.endswith(".zip"):
+            filename += ".zip"
```

Comment:
Is download zipped not zipping the file anymore? Or is the file being zipped but not renamed with the .zip extension?

Reply:
It is being zipped but not renamed with the zip extension

Reply:
I took a closer look at this and the patch you have is the correct route to fix the issue.  HydroShare correctly creates a zip file with the .zip extension when this endpoint is called. Hsclient is creating the filename using the save path and is not using any metadata returned by HydroShare for the filename to save it as.

Reply:
Nice that was kinda what I thought but wanted to check. Thanks for looking at it!